libraries/core_libs/consensus/src/dag/dag_block_manager.cpp

```diff
@@ -192,7 +151,7 @@ std::optional<DagBlock> DagBlockManager::popVerifiedBlock(bool level_limit, uint
       cond_for_verified_qu_.wait(lock);
     }
   }
-  if (stopped_) return std::nullopt;
+  if (stopped_) return {};
```

Comment:
Since there are no more threads in this class maybe we should remove stopped_ variable completely. This is the only place it is used and it could be replaced with the check if the queue is empty. I think it would be cleaner if we actually do always check if the queue is not empty before popping from the queue

Reply:
we can not remove `stopped_` variable, check the code above. `popVerifiedBlock` is called from dag_manager in threads and it is waiting for new block, so this has to stay there until we refactor design of dag_manager. FYI: I wanted to remove it